srcdeps-core-config-yaml/src/main/java/org/srcdeps/config/yaml/writer/YamlWriterVisitor.java

```diff
@@ -63,34 +64,55 @@ public void close() throws IOException {
 
     /** {@inheritDoc} */
     @Override
-    public void containerBegin(ContainerNode<? extends Node> node) {
+    public boolean containerBegin(ContainerNode<? extends Node> node) {
         boolean hasListParent = hasListAncestor(0);
         super.containerBegin(node);
         try {
-            if (!(node instanceof Configuration.Builder) && !node.isInDefaultState(stack)) {
-                indent();
-                if (hasListParent) {
-                    throw new IllegalStateException("Contributions welcome");
-                } else {
-                    out.write(node.getName());
-                    out.write(':');
-                    out.write(configuration.getNewLine());
+            if (!node.isInDefaultState(stack)) {
+                writeComment(node.getCommentBefore());
+                if (!(node instanceof Configuration.Builder)) {
+                    indent();
+                    if (hasListParent) {
+                        throw new IllegalStateException("Contributions welcome");
+                    } else {
+                        out.write(node.getName());
+                        out.write(':');
+                        out.write(configuration.getNewLine());
+                    }
+                    depth++;
                 }
-                depth++;
+                return true;
+            } else {
+                return false;
             }
+
         } catch (IOException e) {
             throw new RuntimeException(e);
         }
     }
 
+    private void writeComment(List<String> commentLines) throws IOException {
+        if (!commentLines.isEmpty() && depth != 0) {
+            out.write(configuration.getNewLine());
+        }
+        for (String line : commentLines) {
+            indent();
+            out.write('#');
```

Comment:
Is it for YAML? If so, can you confirm that comments can be indented? I remember that one of those new formats have problems with the comment char after spaces, but not sure it's YAML.